src/main/java/ToDoTask.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * A class representing a basic <code>Task</code>.
+ */
+public class ToDoTask extends Task {
+    private static final long serialVersionUID = 137301L;
+    
+    /**
+     * Creates an instance of a <code>ToDoTask</code>. The completion status is set to <code>false</code> by default.
+     * 
+     * @param description The description of this <code>Task</code>
+     */
+    public ToDoTask (String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Creates an instance of a <code>ToDoTask</code>. Allows the caller to set its completion status.
+     * 
+     * @param description The description of this <code>Task</code>
+     * @param isDone The completion status of the <code>Task</code>
+     */
+    public ToDoTask (String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+    
+    @Override
+    //Returns a copy of this task but with its completion status marked as done
+    public Task getTaskMarkedAsDone() {
+        return new ToDoTask(description, true);
+    }
+
+    @Override
+    //Returns a copy of this task but with its completion status marked as undone
+    public Task getTaskMarkedUndone() {
```

Comment:
May i ask about the situation of when we are required to use this method? Is it an addition function for the user or...
?

Reply:
It's meant to be an additional feature, for cases where you accidentally marked something as done when it's not actually done. It hasn't been fully implemented though, which is a blunder on my part for having pushed it.